### PR TITLE
Return bad request as ValidationProblemDetails

### PR DESCRIPTION
**Pull request by LockTar**
- Return ValidationProblemDetails as result for 400 errors
- Implement AsActionResult the same way as in ResultExtensions
- Make AsActionResult protected virtual so you can override the implementation.

### CODE REVIEW

File: src/Extensions/src/Eventuous.AspNetCore.Web/Http/CommandHttpApiBase'TResult.cs [hunk 1]
```diff
@@ -46,12 +49,40 @@ protected async Task<ActionResult<TResult>> Handle<TCommand>(TCommand command, C
         return AsActionResult<TAggregate>(result);
     }
 
-    static ActionResult<TResult> AsActionResult<T>(Result result) where T : Aggregate
-        => result is ErrorResult error
+    protected virtual ActionResult<TResult> AsActionResult<T>(Result result) where T : Aggregate {
```
Comment (alexeyzimarev) [inline src/Extensions/src/Eventuous.AspNetCore.Web/Http/CommandHttpApiBase'TResult.cs:52]: Why does it need to be virtual?

File: src/Extensions/src/Eventuous.AspNetCore.Web/Http/ResultExtensions.cs [hunk 2]
```diff
@@ -51,5 +60,18 @@ ActionResult AsProblemResult(int statusCode)
...
+                new ValidationProblemDetails(new Dictionary<string, string[]> { ["Domain"] = [error.ErrorMessage] }) {
+                    Status = statusCode,
+                    Title  = error.ErrorMessage,
+                    Detail = error.Exception?.ToString(),
+                    Type   = error.Exception?.GetType().Name
+                }
+            ) {
+                StatusCode   = Status400BadRequest,
+                ContentTypes = new MediaTypeCollection { ContentTypes.ProblemDetails },
+            };
```
Comment (alexeyzimarev) [inline src/Extensions/src/Eventuous.AspNetCore.Web/Http/ResultExtensions.cs:75]: There seems to be quite a lot of repetition in code, does it mean that extensions or something don't work?

Comment (LockTar) [reply src/Extensions/src/Eventuous.AspNetCore.Web/Http/ResultExtensions.cs:75]: Well, that is what I thought as well. For me it was not clear if some things were done on purpose. 
1. the private methods as sub methods (for an example `AsProblemResult`)
2. `CommandHttpApiBase` and `CommandHttpApiBase<T>` aren't following the `ProblemDetails` setup and weren't using the extensions.

But all methods are slightly different so I'm not sure if we could write 1 extension method for everything. Maybe do this in separate commit?

File: src/Extensions/src/Eventuous.AspNetCore.Web/Http/CommandHttpApiBase'TResult.cs (hunk 1 above)
Comment (LockTar) [reply src/Extensions/src/Eventuous.AspNetCore.Web/Http/CommandHttpApiBase'TResult.cs:52]: We need the option to override it. 
We have the following use case:
The `CommandService` is doing validation with FluentValidation. We're rewriting the result of FluentValidation to a `ValidationProblemDetails`. Our `Kiota` client is picking it up and binds the result to the frontend. So the override looks like this:

```C#
protected override ActionResult<TResult> AsActionResult<T>(Result result)
{
    return result is ErrorResult error
        ? error.Exception switch
        {
            ValidationException => MapValidationExceptionAsValidationProblemDetails(error),
            _ => base.AsActionResult<T>(result)
        }
        : base.AsActionResult<T>(result);
}

private BadRequestObjectResult MapValidationExceptionAsValidationProblemDetails(ErrorResult error)
{
    if (error is null || error.Exception is null || error.Exception is not ValidationException)
    {
        throw new ArgumentNullException(nameof(error), "Exception in result is not of the type `ValidationException`. Unable to map validation result.");
    }

    ValidationException exception = (ValidationException)error.Exception;

    var problemDetails = new ValidationProblemDetails()
    {
        Status = StatusCodes.Status400BadRequest,
        Detail = "Please refer to the errors property for additional details."
    };

    var groupFailures = exception.Errors.GroupBy(v => v.PropertyName);
    foreach (var groupFailure in groupFailures)
    {
        problemDetails.Errors.Add(groupFailure.Key, groupFailure.Select(s => s.ErrorMessage).ToArray());
    }

    return new BadRequestObjectResult(problemDetails);
}
```

**Pull request merged**